plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/PositionDeleteWriter.java

```diff
@@ -91,35 +89,24 @@ public void appendPage(Page page)
         blocks[0] = RunLengthEncodedBlock.create(nativeValueToBlock(VARCHAR, utf8Slice(dataFilePath)), page.getPositionCount());
         blocks[1] = page.getBlock(0);
         writer.appendRows(new Page(blocks));
-
-        writtenData = true;
     }
 
     public Collection<Slice> finish()
     {
-        Collection<Slice> commitTasks = new ArrayList<>();
```

Comment:
Is this handled elsewhere, or are we okay with writing empty delete files?

Reply:
This should never occur due to how `IcebergMergeSink` works. It has a map of partition to deletions, which are created on-demand when there is a row to delete. `PositionDeleteWriter` is created for each entry in the map, thus there will always be at least one row.

I think the empty file handling is a leftover from the old way that deletes were implemented in the engine, before we converted all DML to use the merge machinery.

Reply:
Gotcha, makes sense. Thanks